data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/utils/LogPusher.java

```diff
@@ -0,0 +1,88 @@
+package org.opensearch.dataprepper.plugins.sink.utils;
+
+import io.micrometer.core.instrument.Counter;
+import org.opensearch.dataprepper.plugins.sink.buffer.Buffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.CloudWatchLogsException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InputLogEvent;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+
+import java.util.ArrayList;
+
+public class LogPusher {
+    private final Counter logEventSuccessCounter;
+    private final Counter logEventFailCounter;
+    private final Counter requestSuccessCount;
+    private final Counter requestFailCount;
+    final int retryCount;
+    final long backOffTimeBase;
+    static final Logger LOG = LoggerFactory.getLogger(LogPusher.class);
+    public LogPusher(final Counter logEventSuccessCounter, final Counter logEventFailCounter, final Counter requestSuccessCount, final Counter requestFailCount, final int retryCount, final long backOffTimeBase) {
+        this.logEventSuccessCounter = logEventSuccessCounter;
+        this.logEventFailCounter = logEventFailCounter;
+        this.requestSuccessCount = requestSuccessCount;
+        this.requestFailCount = requestFailCount;
+        this.retryCount = retryCount;
+        this.backOffTimeBase = backOffTimeBase;
+    }
+
+    public boolean pushLogs(final Buffer buffer, final CloudWatchLogsClient cloudWatchLogsClient, final String logGroup, final String logStream) throws InterruptedException {
+        boolean failedPost = true;
+        int failCounter = 0;
+
+        ArrayList<InputLogEvent> logEventList = new ArrayList<>();
+
+        for (byte[] data: buffer.getBufferedData()) {
+            InputLogEvent tempLogEvent = InputLogEvent.builder()
+                    .message(new String(data))
+                    .timestamp(System.currentTimeMillis())
```

Comment:
Is this value correct? This sets the timestamp to the current time - should it be the time of the event itself?

Reply:
This value is currently more of a default, if the event itself has it, then I can sub it in. I will try to add this feature in for the next commit.